src/adapters/MeetingsJSONAdapter.test.js

```diff
@@ -1,57 +1,133 @@
+      meetingID = 'localAudio';
+
+      subscription = meetingsJSONAdapter
+        .getMeeting(meetingID)
+        .pipe(tap(() => meetingsJSONAdapter.toggleMuteAudio(meetingID)))
+        .subscribe((meeting) => {
+          expect(meeting.localAudio.muted).toBeTruthy();
+          done();
+        });
+    });
+
+    test('throws a proper error message', (done) => {
+      const wrongMeetingID = 'wrongMeetingID';
+
+      subscription = meetingsJSONAdapter.getMeeting(wrongMeetingID).subscribe(
+        () => {},
+        (error) => {
+          expect(error.message).toBe(`Could not find meeting with ID "${wrongMeetingID}"`);
+          done();
+        }
+      );
+    });
   });
 
-  test('getMeeting() completes the observable', (done) => {
-    meetingsJSONAdapter.getMeeting(meetingID).subscribe(
-      () => {},
-      () => {},
-      () => {
-        expect(true).toBeTruthy();
-        done();
-      }
-    );
+  describe('muteAudioControl() returns', () => {
+    beforeEach(() => {
+      meetingID = 'localAudio';
+    });
+
+    test('active control display values', (done) => {
+      rxjs.fromEvent = jest.fn(() => from([{detail: meetings[meetingID]}]));
+
+      subscription = meetingsJSONAdapter
+        .muteAudioControl(meetingID)
+        .pipe(skip(1)) // Skip the "default" emission
+        .subscribe((display) => {
+          expect(display).toMatchObject({
+            ID: 'mute-audio',
+            icon: 'microphone-muted',
+            tooltip: 'Unmute',
+            state: 'active',
+          });
+          done();
+        });
+    });
+
+    test('inactive control display values', (done) => {
+      rxjs.fromEvent = jest.fn(() => {
+        const meeting = meetings[meetingID];
+
+        meeting.localAudio.muted = false; // Local audio is already muted
+
+        return from([{detail: meeting}]);
+      });
+
+      subscription = meetingsJSONAdapter
+        .muteAudioControl(meetingID)
+        .pipe(skip(1)) // Skip the "default" emission
+        .subscribe((display) => {
+          expect(display).toMatchObject({
+            ID: 'mute-audio',
+            icon: 'microphone',
+            tooltip: 'Mute',
+            state: 'inactive',
+          });
+          done();
+        });
+    });
+
+    test('error on invalid meeting ID', (done) => {
+      subscription = meetingsJSONAdapter.muteAudioControl('invalid').subscribe(
+        () => {},
+        (error) => {
+          expect(error.message).toEqual('Could not find meeting with ID "invalid"');
+          done();
+        }
+      );
+    });
   });
 
-  test('getMeeting() renders the local media if localVideo property is defined', (done) => {
-    meetingsJSONAdapter.getMeeting('localVideo').subscribe((data) => {
-      expect(data.localVideo).toEqual('mock-stream');
-      done();
+  describe('toggleMuteAudio()', () => {
+    test('dispatches a "mute-audio" event', () => {
+      meetingID = 'localAudio';
```

Comment:
we are exactly doing something that you are claiming the opposite. we are setting `meetingID` to something first and then set it to something else for this test case.